### PR TITLE
feat: logging improvement

### PR DESCRIPTION
**Pull request by JanLewDev**
#67

### CODE REVIEW

File: packages/network/src/node.ts [hunk 1]
```diff
@@ -25,12 +25,33 @@ import { webTransport } from "@libp2p/webtransport";
...
+import { Logger, ILogObj, ISettingsParam } from "tslog";
+
+let mainLogger: Logger<ILogObj> = new Logger();
+
+let startLogger: Logger<ILogObj> = new Logger();
+let subscribeLogger: Logger<ILogObj> = new Logger();
+let unsubscribeLogger: Logger<ILogObj> = new Logger();
+let broadcastMessageLogger: Logger<ILogObj> = new Logger();
+let sendMessageLogger: Logger<ILogObj> = new Logger();
+let sendGroupMessageRandomPeerLogger: Logger<ILogObj> = new Logger();
```
Comment (d-roak) [inline packages/network/src/node.ts:37]: have only, one logger per package. Prepend all the messages from ea pkg with it's name: `topology::{pkg_name}`

File: packages/network/src/node.ts [hunk 2]
```diff
@@ -40,8 +61,11 @@ export class TopologyNetworkNode {
 
   peerId: string = "";
 
-  constructor(config?: TopologyNetworkNodeConfig) {
+  constructor(config: TopologyNetworkNodeConfig) {
     this._config = config;
+    mainLogger = new Logger(config.tslogconfig);
+    mainLogger.settings.name = "topology:network";
```
Comment (d-roak) [inline packages/network/src/node.ts:67]: yep, like this !! do "::" instead

File: packages/node/src/index.ts [hunk 3]
```diff
@@ -10,12 +10,13 @@ import { TopologyObjectStore } from "./store";
...
 import { toString as uint8ArrayToString } from "uint8arrays/to-string";
 import { OPERATIONS } from "./operations.js";
+import tslogconfig from "../tslog.config";
 
 export * from "./operations.js";
 
 // snake_casing to match the JSON config
 export interface TopologyNodeConfig {
-  network_config?: TopologyNetworkNodeConfig;
+  network_config: TopologyNetworkNodeConfig;
```
Comment (d-roak) [inline packages/node/src/index.ts:19]: why?

File: packages/network/src/node.ts [hunk 4]
```diff
@@ -25,12 +25,33 @@ import { webTransport } from "@libp2p/webtransport";
...
+  sendGroupMessageRandomPeerLogger = mainLogger.getSubLogger({ name: "sendGroupMessageRandomTopicPeer" });
+}
+
 // snake_casing to match the JSON config
 export interface TopologyNetworkNodeConfig {
   addresses?: string[];
   bootstrap?: boolean;
   bootstrap_peers?: string[];
   private_key_seed?: string;
+  tslogconfig: ISettingsParam<ILogObj>;
```
Comment (d-roak) [inline packages/network/src/node.ts:54]:
```suggestion
  tslog_config: ISettingsParam<ILogObj>;
```

Comment (d-roak) [reply packages/network/src/node.ts:54]: this is setted up with json files, what are the options for it? did you tested with the config files?

File: tsconfig.json [hunk 5]
```diff
@@ -14,6 +14,7 @@
 
     "skipLibCheck": true,
     "stripInternal": true,
-    "resolveJsonModule": true
+    "resolveJsonModule": true,
+    "inlineSourceMap": true
```
Comment (d-roak) [inline tsconfig.json:18]: why?

File: packages/node/tslog.config.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { ILogObj, ISettingsParam } from "tslog"
+
+// https://tslog.js.org/#/?id=settings
+const tslogconfig: ISettingsParam<ILogObj> = {
+    minLevel: 3, // 3 = INFO
+    type: "hidden", // "pretty" | "json" | "hidden" 
+    hideLogPositionForProduction: true,
+};
+
+export default tslogconfig;
```
Comment (d-roak) [inline packages/node/tslog.config.ts:10]: should be in the existing config files

File: packages/node/tsconfig.json [hunk 7]
```diff
@@ -14,5 +14,5 @@
       "path": "../object"
     }
   ],
-  "include": ["src/**/*.ts"]
+  "include": ["src/**/*.ts", "./*.config.ts"]
```
Comment (d-roak) [inline packages/node/tsconfig.json:17]: we are dropping the `tslog.config.ts`, you can revert this

File: packages/node/src/index.ts [hunk 8]
```diff
@@ -10,12 +10,13 @@ import { TopologyObjectStore } from "./store";
 import { fromString as uint8ArrayFromString } from "uint8arrays/from-string";
 import { toString as uint8ArrayToString } from "uint8arrays/to-string";
 import { OPERATIONS } from "./operations.js";
+import tslogconfig from "../tslog.config";
 
 export * from "./operations.js";
 
 // snake_casing to match the JSON config
 export interface TopologyNodeConfig {
```
Comment (d-roak) [inline packages/node/src/index.ts:18]: node also has a tslog_config that can be different from the network config

by default, the network will be equal to the node, but they can be set to be different

**Review by d-roak — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 5 above)
Comment (JanLewDev) [reply tsconfig.json:18]: image
so the line is referenced correctly

File: packages/network/src/node.ts (hunk 4 above)
Comment (JanLewDev) [reply packages/network/src/node.ts:54]: should it be a new json file or add to tsconfig.json?

File: packages/node/tslog.config.ts (hunk 6 above)
Comment (JanLewDev) [reply packages/node/tslog.config.ts:10]: tsconfig.json, right?

Comment (JanLewDev) [reply packages/node/tslog.config.ts:10]: so the tsconfig.json for node would look something like this?
```typescript
{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "dist"
  },
  "references": [
    {
      "path": "../crdt"
    },
    {
      "path": "../network"
    },
    {
      "path": "../object"
    }
  ],
  "include": ["src/**/*.ts"],
  "tslog_config": {
    "minLevel": 3, // 3 = INFO
    "type": "pretty", // "pretty" | "json" | "hidden" 
    "hideLogPositionForProduction": true
  }
}
```

File: packages/network/src/node.ts [hunk 9]
```diff
@@ -25,12 +25,17 @@ import { webTransport } from "@libp2p/webtransport";
 import { autoNAT } from "@libp2p/autonat";
 import { fromString as uint8ArrayFromString } from "uint8arrays/from-string";
 
+import { Logger, ILogObj, ISettingsParam } from "tslog";
+
+let mainLogger: Logger<ILogObj> = new Logger();
```
Comment (d-roak) [inline packages/network/src/node.ts:30]: rename it to `log`, so we can use `log.info()`

File: packages/network/src/node.ts [hunk 10]
```diff
@@ -25,12 +25,17 @@ import { webTransport } from "@libp2p/webtransport";
 import { autoNAT } from "@libp2p/autonat";
 import { fromString as uint8ArrayFromString } from "uint8arrays/from-string";
```
Comment (d-roak) [inline packages/network/src/node.ts:27]:
```suggestion
```

File: packages/node/configs/example.json [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+        // Minimum log level to log. 
+        // Options: 0: silly, 1: trace, 2: debug, 3: info, 4: warn, 5: error, 6: fatal
+        "minLevel": 3,
+
+        // Type - hidden || json || pretty
+        "type": "hidden",
+
+        // Hide log position for production
+        "hideLogPositionForProduction": true
+      }
```
Comment (d-roak) [inline packages/node/configs/example.json:24]: only options that exist?

File: packages/node/tslog.config.ts (hunk 6 above)
Comment (d-roak) [reply packages/node/tslog.config.ts:10]: are you still using this?

File: packages/node/src/index.ts [hunk 12]
```diff
@@ -26,7 +28,19 @@ export class TopologyNode {
 
   constructor(config?: TopologyNodeConfig) {
     this._config = config;
-    this.networkNode = new TopologyNetworkNode(config?.network_config);
+    if(!this._config?.tslog_config){
+      if(!this._config) this._config = {};
```
Comment (d-roak) [inline packages/node/src/index.ts:32]: this shouldn't be inside the prev if

**Review by d-roak — CHANGES_REQUESTED** [review]

**Review by d-roak — APPROVED** [review]
lgtm

**Pull request merged**